frontend/projects/upgrade/src/app/features/dashboard/segments/pages/segment-details-page/segment-details-page-content/segment-lists-section-card/segment-lists-section-card.component.ts

```diff
@@ -65,16 +67,39 @@ export class SegmentListsSectionCardComponent {
     private authService: AuthService
   ) {}
 
+  private segmentTypeConfig = {
+    [SEGMENT_TYPE.GLOBAL_EXCLUDE]: {
+      title: 'segments.details.lists.card.title.global-excludes.text',
+      subtitle: 'segments.details.lists.card.subtitle.global-excludes.text',
+      buttonText: 'segments.details.add-exclude-list.button.text',
+      importLabel: 'segments.details.lists.menu-button.import-exclude-list.text',
+      exportLabel: 'segments.details.lists.menu-button.export-exclude-lists.text',
+    },
+    [SEGMENT_TYPE.PUBLIC]: {
+      title: 'segments.details.lists.card.title.text',
+      subtitle: 'segments.details.lists.card.subtitle.text',
+      buttonText: 'segments.details.add-list.button.text',
+      importLabel: 'segments.details.lists.menu-button.import-list.text',
+      exportLabel: 'segments.details.lists.menu-button.export-lists.text',
+    },
+  };
+
   ngOnInit() {
     this.permissions$ = this.authService.userPermissions$;
-    this.title =
-      this.data.type === SEGMENT_TYPE.GLOBAL_EXCLUDE
-        ? 'segments.details.lists.card.title.global-excludes.text'
-        : 'segments.details.lists.card.title.text';
+    this.configureUiForSegmentType();
+  }
+
+  private configureUiForSegmentType(): void {
+    const config = this.segmentTypeConfig[this.data.type];
+    this.title = config.title;
+    this.subtitle = config.subtitle;
+    this.buttonText = config.buttonText;
+    this.menuButtonItems[0].label = config.importLabel;
+    this.menuButtonItems[1].label = config.exportLabel;
   }
 
   onAddListClick(appContext: string, segmentId: string) {
-    this.dialogService.openAddListModal(appContext, segmentId);
+    this.dialogService.openAddListModal(appContext, segmentId, this.data.type);
```

Comment:
for all of these, you can pass in a modal title directly instead of making the modal figure it out based passing in a type, you've got a convenient object `segmentTypeConfig` that is doing that kind of thing already.